kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/KafkaProxyContext.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.operator;
+
+import java.time.Clock;
+import java.util.stream.Collectors;
+
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxy;
+import io.kroxylicious.kubernetes.api.v1alpha1.VirtualKafkaCluster;
+import io.kroxylicious.kubernetes.operator.model.ProxyModel;
+import io.kroxylicious.kubernetes.operator.model.ProxyModelBuilder;
+
+public class KafkaProxyContext {
+
+    private static final String KEY_SEC = "sec";
+    private static final String KEY_MODEL = "model";
+    private static final String KEY_CLOCK = "clock";
+
+    private KafkaProxyContext() {
+    }
+
+    static void init(Clock clock,
+                     SecureConfigInterpolator secureConfigInterpolator,
+                     KafkaProxy proxy,
+                     Context<KafkaProxy> context) {
+        var rc = context.managedWorkflowAndDependentResourceContext();
+
+        rc.put(KEY_CLOCK, Clock.fixed(clock.instant(), clock.getZone()));
+
+        rc.put(KEY_SEC, secureConfigInterpolator);
+
+        ProxyModelBuilder proxyModelBuilder = ProxyModelBuilder.contextBuilder();
+        ProxyModel model = proxyModelBuilder.build(proxy, context);
```

Comment:
this feels a little hidden away that a `*Context` named class does some useful modelling work in an init method, I think I'd push the ProxyModel in from KPReconciler so it's more visible that we're building the model there